### PR TITLE
[examples] Add Vercel Analytics to Astro template

### PR DESCRIPTION
**Pull request by joshuadavidthomas**
This PR ports Vercel Analytics over from the Sveltekit example, adapted for Astro.

The version of `web-vitals` has been bumped to the most recent version, and as a consequence the function imports from that package have been changed, as it appears `get<Vital>` has been depreciated in favor of `on<Vital>`.

### CODE REVIEW

File: examples/astro/src/layouts/Layout.astro [hunk 1]
```diff
@@ -14,6 +14,19 @@ const { title } = Astro.props as Props;
 		<link rel="icon" type="image/x-icon" href="/favicon.ico" />
 		<meta name="generator" content={Astro.generator} />
 		<title>{title}</title>
+		<script>
+			import { webVitals } from "../lib/vitals";
+
+			let analyticsId = import.meta.env.PUBLIC_VERCEL_ANALYTICS_ID;
```
Comment (leerob) [inline examples/astro/src/layouts/Layout.astro:20]: Have you tested this works? 🙏

**Review by leerob — APPROVED** [review]
Code seems fine - thank you!

File: examples/astro/src/layouts/Layout.astro (hunk 1 above)
Comment (joshuadavidthomas) [reply examples/astro/src/layouts/Layout.astro:20]: Yep! I actually ported it initially for the site I'm working on for work, then forked and added it to this repo. 

Screenshots:

image

image

Comment (joshuadavidthomas) [reply examples/astro/src/layouts/Layout.astro:20]: [Relevant section in Astro docs](https://docs.astro.build/en/core-concepts/astro-components/#client-side-scripts):

> By default, <script> tags are processed by Astro.
>
> Any imports will be bundled, allowing you to import local files or Node modules.
> -  The processed script will be injected into your page’s <head> with type="module".
> - TypeScript is fully supported, including importing TypeScript files
> - If your component is used several times on a page, the script tag will only be included once.
>
> ```html
> <script>
>   // Processed! Bundled! TypeScript-supported! ESM imports work, even to npm packages.
> </script>
> ```

**Review by leerob — APPROVED** [review]

**Comment by joshuadavidthomas** [comment]
@leerob Anything I can do to help push this over the finish line? It appears to me all the failing checks are because I'm not a Vercel employee.

**Comment by EndangeredMassa** [comment]
Thanks @joshuadavidthomas!

**Pull request merged**